lib/request.ts

```diff
@@ -45,18 +76,26 @@ export const request = (options: ServiceClientRequestOptions): Promise<ServiceCl
   }
 
   const httpRequestFn = options.protocol === 'https:' ? httpsRequest : httpRequest
-  return new Promise((resolve, reject) => {
+  return new Promise((resolve, originalReject) => {
     let hasRequestEnded = false
     let startTime: [number, number]
     let timings: Timings
+    let reject = originalReject
     if (options.timing) {
       startTime = process.hrtime()
       timings = {
-        lookup: -1,
```

Comment:
I changed `-1` to `undefined`. Should we mark this as a breaking change?

Reply:
I would say that it is a breaking change. What is the reasoning for it?

Reply:
- -1 will skew metrics if users monitor timings with average, percentiles, etc.
- The code to fill timings.lookup and timings.connect in response handler is not working because it checks `undefined`.
- -1 in timings causes invalid numbers in timingPhases. For example, if `response` is 100 and `end` is -1, `download` becomes -99, which is totally wrong. This can happen in timeout errors.

To avoid the issues above, users and `request.ts` need to treat `-1` as `undefined` and ignore it properly. If we document the `-1` as virtual `undefined`, timings users need to change their code to treat `-1` as `undefined`. Then I think it makes more sense to have `undefined` instead of `-1` if the users need to change their code anyway.